src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -1160,6 +1162,39 @@ static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x,
 
 } // namespace BinaryIntrinsicFunction
 
+
+namespace ObjectType {
+
+     static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x, diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args != 1,
+            "ASR Verify: type() takes only 1 argument `object`",
+            x.base.base.loc, diagnostics);
+    }
+
+    static ASR::expr_t *eval_ObjectType(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*>& /*args*/) {
+        std::string object_type = "<type '" + ASRUtils::type_to_str_python(t1) + "'>";
+        return StringConstant(object_type, character(object_type.length()));
+    }
+
+    static inline ASR::asr_t* create_ObjectType(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() != 1) {
+            err("type() takes exactly 1 argument `object` for now", loc);
+        }
+
+        Vec<ASR::expr_t *> arg_values;
+        ASR::expr_t *m_value = eval_ObjectType(al, loc, expr_type(args[0]), arg_values);
+        return ASR::make_IntrinsicScalarFunction_t(al, loc,
+            static_cast<int64_t>(IntrinsicScalarFunctions::ObjectType),
+            args.p, args.n, 0, ASRUtils::expr_type(args[0]), m_value);
```

Comment:
```suggestion
            args.p, args.n, 0, ASRUtils::expr_type(m_value), m_value);
```

Reply:
This fails withe the ASR Verify error stated above.

Reply:
Apply my other suggestions, the ASR Verify error would be fixed.

Reply:
I am in the process 😃.

Reply:
@Thirumalai-Shaktivel I am interested to know the reason this worked. I guess it was because of this line - `{nullptr, &ObjectType::verify_args}}`?

Reply:
Yes! It's because of the function `UnaryIntrinsicFunction::verify_args` which is used for the intrinsics that have same type for the input and output.